resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+                    String effectiveTime = resourcePathSegments[2];
+                    if (dependentTargetEffectiveTimes.contains(effectiveTime)) {
+                        possibleRF2Packages.add(availableRF2Package);
+                    }
+                }
+
+                if (possibleRF2Packages.isEmpty()) {
+                    continue;
+                }
+
+                for (String possibleRF2PackagePath : possibleRF2Packages) {
+                    File possibleRF2Package = resourceManager.doReadResourceFile(possibleRF2PackagePath);
+                    Set<String> uniqueModuleIds = rf2Service.getUniqueModuleIds(possibleRF2Package);
+                    boolean owningPackageFound = uniqueModuleIds.contains(rf2Row.getColumn(RF2Service.REFERENCED_COMPONENT_ID));
+                    if (owningPackageFound) {
+                        rf2Row.setFound(true);
+                        found = found + 1;
+                        rf2Row.setMetadataResourcePath(asMetadataResourcePath(possibleRF2PackagePath));
+                    }
+                }
+            }
+
+            if (found == rf2Rows.size()) {
+                break;
+            }
+        }
+
+        Set<String> metadataResourcePaths = new HashSet<>();
+        for (RF2Row rf2Row : rf2Rows) {
+            if (!rf2Row.isFound()) {
+                throw new RuntimeException("Cannot generate metadata; missing dependency");
+            } else {
+                metadataResourcePaths.add(rf2Row.getMetadataResourcePath());
+            }
+        }
+
+        Set<ModuleMetadata> moduleMetadatas = new TreeSet<>(Comparator.comparingInt(ModuleMetadata::getEffectiveTime));
+        for (String dep : metadataResourcePaths) {
+            File file = resourceManager.doReadResourceFile(dep);
+            ModuleMetadata moduleMetadata = FileUtils.convertToObject(file, ModuleMetadata.class);
+            moduleMetadatas.add(moduleMetadata);
+        }
+
+        return moduleMetadatas;
+    }
+
+    private Integer asInteger(String value) {
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String asMetadataResourcePath(String packageResourcePath) {
+        String[] splits = packageResourcePath.split("/");
+        splits = Arrays.copyOf(splits, splits.length - 1); // Remove last segment, i.e. a/b/c/d => a/b/c
+
+        return String.join("/", splits) + "/metadata.json";
+    }
+
+    private ModuleMetadata doDownloadRelease(String codeSystem, String moduleId, String effectiveTime, boolean includeFile) {
+        for (String readDirectory : this.readDirectories) {
+            LOGGER.trace("Downloading {}/{}_{}/{}", readDirectory, codeSystem, moduleId, effectiveTime);
+            String baseResourcePath = getBaseResourcePath(readDirectory, codeSystem, moduleId, effectiveTime);
```

Comment:
@pgwilliams Recursively read from directories, i.e. if not found in Dev, then try Prod.